pliers/extractors/audio.py

```diff
@@ -618,3 +618,55 @@ def _extract(self, stim):
         return ExtractorResult(preds, stim, self, features=labels,
                                onsets=onsets, durations=[dur]*len(onsets),
                                orders=list(range(len(onsets))))
+
+
+class MFCCEnergyExtractor(AudioExtractor):
+    ''' Given a Middle Bound, Extract the Energy from the Low and 
+    High Registers of Mel-Frequency Ceptral Coefficients.
+
+    Args:
+    n_mfcc (int): specifies the number of MFCC to extract
+    n_coefs(int): cepstrum coefficients to keep in low quefrency spectrum
+    hop_length (int): hop length in number of samples
+    librosa_kwargs (optional): Optional named arguments to pass to librosa
+    '''
+
+    _log_attributes = ('hop_length', 'n_coefs', 'n_mfcc')
+
+    def __init__(self, n_mfcc=48, n_coefs=13, hop_length=1024, **librosa_kwargs):
+        self.n_mfcc = n_mfcc
+        self.n_coefs = n_coefs
+        self.hop_length = hop_length
+        self.librosa_kwargs = librosa_kwargs
+        super().__init__()
+
+    def _extract(self, stim):
+
+        #https://librosa.org/doc/main/generated/librosa.feature.mfcc.html
+        mfccs = librosa.feature.mfcc(y=stim.data, 
+            sr=stim.sampling_rate, 
+            n_mfcc=self.n_mfcc,
+            hop_length=self.hop_length,
+            **self.librosa_kwargs)
+
+        #https://docs.scipy.org/doc/scipy/reference/generated/scipy.fft.dct.html#scipy.fft.dct
+        low_mfs = fft.dct(np.transpose(mfccs[:self.n_coefs]), 
+            type=2, 
+            n=self.n_mfcc, 
+            axis=- 1, 
+            norm='ortho', 
+            overwrite_x=False)
+        high_mfs = fft.dct(np.transpose(mfccs[self.n_coefs:]), 
+            type=2, 
+            n=self.n_mfcc, 
+            axis=- 1, 
+            norm='ortho', 
+            overwrite_x=False)
+
+        low_mfs = 10 ** (low_mfs / 20.)
+        high_mfs = 10 ** (high_mfs / 20.)
+
+        out_val = {'low':low_mfs,'high':high_mfs}
+
```

Comment:
To allow the pliers `to_df` method to behave appropriately we need to define a `get_feature_names` function. 

This is an example (I have not verified this code);

```suggestion
def get_feature_names():
    return [f"{name}_{i}" for name in ['low', 'high'] for i in range(0, self.n_n_mfcc)]   
```
if `n_mfcc` it should return the following:
```
['low_0',
 'low_1',
 'low_2',
 'low_3',
 'low_4',
 'low_5',
 'low_6',
 'low_7',
 'low_8',
 'low_9',
 'high_0',
 'high_1',
 'high_2',
 'high_3',
 'high_4',
 'high_5',
 'high_6',
 'high_7',
 'high_8',
 'high_9']
```
Thus, the 2-D matrix that is return to `ExtractorResult` should be (number of frames) x (n_mfcc * 2) (for each low and high).